components/scrubber/scrubber.go

```diff
@@ -48,8 +48,22 @@ type Scrubber interface {
 	Struct(val any) error
 }
 
-// Scrub scrubs the implementing type from sensitive information
-type Scrub interface {
+// Scrubbable is the interface that must be implemented by types that can be scrubbed
+// from sensitive information. The Scrub method is called with a Scrubber that can be used
+// to scrub the type.
+//
+// Example:
+//
+//	  type Example struct {
+//			Username      string
+//		    Password      string
+//	  }
+//
+//	  func (e *Example) Scrub(scrubber Scrubber) error {
```

Comment:
You mean it would be better if it returns a value? So it would be up to a client to decide whether to copy or mutate in the place?

Reply:
It is actually complicated because how we scrub things. Scrubbing does mutate the structure in general now.

I think maybe we can use another approach similar to what VS Code does and introduce:
```golang
type TrustedValue struct {
    value any
}
```

Scrubber will trust that client code already scrubbed an underlying value in the appropriate way and use such value.

Reply:
Talked with @csweichel that it should be rather an interface.

Reply:
@iQQBot I pushed new changes, does it make more sense to you?

Reply:
not necessary, it can be the same struct